### DISCUSSION TITLE
Failed to install RPiR

### DISCUSSION THREAD

**Discussion by XinyueJia** [IBAHCM/RPiR#130] [Archive] [answered]
I am struggling with RPiR installation. It seems like my local network problems. Is there any other way to install it (.tgz)? 

<img width="615" alt="Screen Shot 2021-01-23 at 1 57 36 AM" src="https://user-images.githubusercontent.com/77334755/105530006-325a9300-5d22-11eb-84c2-fb4ed5f027e2.png">

**Comment by soniamitchell** [comment]
Hi @Xinyue-Jia, could you try restarting your computer and trying again? 

If that doesn't work, try adding a [PAT](https://ibahcm.github.io/RPiR/articles/pages/github.html#add-a-pat), restarting your computer again, and then running `install_github("IBAHCM/RPIR", force = TRUE)`.

**Reply by XinyueJia** [reply]
Thank you @soniamitchell , but it still didn't work.

<img width="602" alt="Screen Shot 2021-01-23 at 7 57 38 PM" src="https://user-images.githubusercontent.com/77334755/105577548-5f0bba80-5db5-11eb-9038-e3e78753d64f.png">


Also, I downloaded the '.tar' file according to its url and then attempted to install it from local file.
<img width="709" alt="Screen Shot 2021-01-23 at 6 57 47 PM" src="https://user-images.githubusercontent.com/77334755/105576456-d1789c80-5dad-11eb-9fb4-5f94b98d8ace.png">
It doesn't seem to work either...
I am currently at home (China), so I thought it is possibly caused by my local network situation ('Great Firewall'). I tried to use VPN or change the DNS but useless.

**Reply by soniamitchell** [reply]
Hi @Xinyue-Jia, this is mysterious... 

I don't know what `Error exit delayed from previous errors` means... perhaps you could try restarting Rstudio / your computer and trying again?

Otherwise, it seems that you're having problems installing packages from GitHub. So if it is related to the firewall (this is beyond my expertise, sorry) I'd recommend making a note of the dependencies and installing them separately?

You need to get these from GitHub:
https://api.github.com/repos/rstudio/learnr/tarball/HEAD
https://api.github.com/repos/rstudio-education/gradethis/tarball/HEAD

Then install everything else that usually comes up when you try `install_github()`, so from the above example:
```
install.packages(c("colorspace", "crosstalk", "isoband", "DT", "ggplot2", "dplyr"))
```

Then install RPiR without dependencies (so that learnr and gradethis don't break the install):
```
install_local("~/Downloads/IBAHCM-RPiR-dedfd6b.tar", dependencies = FALSE)
```

**Comment by richardreeve** [comment, marked as answer]
Hi @Xinyue-Jia, the error with the truncated tar archive is pretty straightforward I think. You just haven't downloaded the whole file. In fact all of this seems to be associated with a poor connection to api.github.com. If it doesn't fix itself when you try to reinstall now, then I would try to download these four files:

1. https://github.com/rstudio/htmltools/archive/master.zip
2. https://github.com/rstudio/learnr/archive/master.zip
3. https://github.com/rstudio-education/gradethis/archive/master.zip
4. https://github.com/IBAHCM/RPiR/archive/master.zip

And then one at a time:

- unzip it into some handy folder on your computer
- In Rstudio, **Open Project in New Session...** and select the unzipped folder
- Run `devtools::install()` in the console, and if you get asked what updates to make, select 2: CRAN packages only
- Close the RStudio instance you're using

So four times through that loop. If this works, then for future updates, you'll only have to update RPiR, and it's a pain, but for some reason your computer doesn't seem to like github... anyway, let me know how it goes?

**Reply by soniamitchell** [reply]
Definitely pressing 2 is the superior option for installing CRAN dependencies, do this instead of `install.packages()`. Sorry, I forgot it existed! (usually just blindly press 1)

**Reply by XinyueJia** [reply]
Thanks !@soniamitchell @richardreeve , somehow I just found `devtools::install_github()` worked, and I managed to install RPiR. Yet, the connection is still weak and unstable, I was wondering would there be other Github packages involved in the future classes? I think it would be better to download their local files in advance...anyway, thanks a lot for your help!

**Reply by soniamitchell** [reply]
Hey @Xinyue-Jia, that's great news! As far as I'm aware, that's everything from GitHub, so you shouldn't have to worry :)

**Reply by XinyueJia** [reply]
Thank you!

**Reply by richardreeve** [reply]
Yes, great news @Xinyue-Jia! You will need to reinstall `RPiR` several more times as the course is updated, and if `install_github()` stops working again, you can try my suggestion above, but you won't need to download all of the files, only https://github.com/IBAHCM/RPiR/archive/master.zip now it's working once.
